### DISCUSSION TITLE
Does anyone know how to increase the "source documents return" - k?

### DISCUSSION THREAD

**Discussion by dkampien** [mayooear/ai-pdf-chatbot-langchain#188] [Q&A]
In the updated repo, the makechain changed, there is no K variable anymore.
![image](https://user-images.githubusercontent.com/32043605/233358080-42719593-f1da-4bda-a888-c6f6a1cd93ff.png)

**Comment by Haste171** [comment]
This is the old code which uses a different function, not sure if you can with the new function logic it uses...
![image](https://user-images.githubusercontent.com/34923485/233762316-0d49fe71-d578-4108-b3fe-cb65dffca41b.png)

**Comment by palef-re** [comment]
interested in this - if you can't set a higher K value, is there a way to tweak the logic so more source documents are generally returned?

**Comment by flickersgit** [comment]
Here,

Put the number you desired in the 'as.retriever'.

  ```
const chain = ConversationalRetrievalQAChain.fromLLM(
    model,
    vectorstore.asRetriever(12),
    {
      qaTemplate: QA_PROMPT,
      questionGeneratorTemplate: CONDENSE_PROMPT,
      returnSourceDocuments: true,  
      //The number of source documents returned is 4 by default
    },
  );
  return chain;
```

**Reply by amitaknkpt** [reply]
Yes

> Here,
> 
> Put the number you desired in the 'as.retriever'.
> 
> ```
> const chain = ConversationalRetrievalQAChain.fromLLM(
>   model,
>   vectorstore.asRetriever(12),
>   {
>     qaTemplate: QA_PROMPT,
>     questionGeneratorTemplate: CONDENSE_PROMPT,
>     returnSourceDocuments: true,  
>     //The number of source documents returned is 4 by default
>   },
> );
> return chain;
> ```

It worked for me

**Comment by ahhany** [comment]
if the number is high like vectorstore.asRetriever(40) the interface shows "Request failed with status code 400" , is there a way to change the interface to be similar to the chatgpt writing the response as it gets. and how can I get detailed response from my files if the response should be an analysis not just a simple question?

**Comment by jon-fuller-ukhsa** [comment]
Is there an equivalent but for Python?

**Reply by onalv** [reply]
seems to be like this as per this link [Stackoverflow](https://stackoverflow.com/questions/76624914/how-do-i-change-the-default-4-documents-that-langchain-returns#comment135161042_76624914)

`In Python you can pass search_kwargs param to as_retriever() like: as_retriever(search_kwargs={"k": 3}). There could be a similar interface in JS binding probably.`
